hazelcast/src/test/java/com/hazelcast/scheduledexecutor/ScheduledExecutorServiceSlowTest.java

```diff
@@ -103,23 +103,26 @@ public void stats_longRunningTask_durable()
 
         String key = generateKeyOwnedBy(instances[1]);
 
-        ICountDownLatch latch = instances[0].getCountDownLatch("latch");
-        latch.trySetCount(6);
+        ICountDownLatch firstLatch = instances[0].getCountDownLatch("firstLatch");
+        firstLatch.trySetCount(2);
+
+        ICountDownLatch lastLatch = instances[0].getCountDownLatch("lastLatch");
+        lastLatch.trySetCount(6);
 
         IScheduledExecutorService executorService = getScheduledExecutor(instances, "s");
         IScheduledFuture future = executorService.scheduleOnKeyOwnerAtFixedRate(
-                new ICountdownLatchRunnableTask("latch"), key, 0, 10, SECONDS);
+                new ICountdownLatchRunnableTask("firstLatch", "lastLatch"), key, 0, 10, SECONDS);
 
-        Thread.sleep(12000);
+        firstLatch.await(12, SECONDS);
 
         instances[1].getLifecycleService().shutdown();
 
-        latch.await(70, SECONDS);
+        lastLatch.await(70, SECONDS);
         sleepSeconds(4); // Wait for run-cycle to finish before cancelling, in order for stats to get updated.
         future.cancel(false);
 
         ScheduledTaskStatistics stats = future.getStats();
-        assertEquals(6, stats.getTotalRuns());
+        assertEquals(6, stats.getTotalRuns(), 1);
```

Comment:
@tkountis I understand you cannot eliminate completely the need for some wait for updating stats, but you can avoid the major source of uncertainty by submitting something like this task:
```
static class GatedCountdownLatchRunnable extends ICountdownLatchRunnableTask {

        final int executions;
        final String counterName;

        GatedCountdownLatchRunnable(int executions, String... runsCounterlatchNames) {
            super(runsCounterlatchNames);
            this.executions = executions;
            this.counterName = randomString();
        }

        @Override
        public void run() {
            IAtomicLong counter = instance.getAtomicLong(counterName);
            if (counter.get() < executions) {
                for (String runsCounterLatchName : runsCounterlatchNames) {
                    instance.getCountDownLatch(runsCounterLatchName).countDown();
                }
                counter.getAndIncrement();
            }
        }
    }
```

@pveentjer I was refering to the future as in wall-clock time sense, not a specific `Future` ;-)